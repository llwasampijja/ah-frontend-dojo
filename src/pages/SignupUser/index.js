// react libraries
import React, { Component } from 'react';

// components
import SignupForm from 'components/SignupForm';

// actions
import RegisterUser from 'store/actions/registerUser';

// store
import store from 'store/store';

class SignupUser extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      username: '',
      password: '',
      confirmPassword: '',
    };
  }

  onSubmitHandler = (event) => {
    event.preventDefault();
    const {
      email, username, password,
    } = this.state;
    const signupData = {
      user: {
        username,
        email,
        password,
      },
    };
    store.dispatch(RegisterUser(signupData));
  }

  onChangeHandler = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  };

  render() {
    const {
      email, username, password, confirmPassword,
    } = this.state;
    return (
      <div>
        <SignupForm
          email={email}
          username={username}
          password={password}
          confirmPassword={confirmPassword}
          onChangeHandler={this.onChangeHandler}
          onSubmitHandler={this.onSubmitHandler}
        />
      </div>
    );
  }
}

export default SignupUser;
