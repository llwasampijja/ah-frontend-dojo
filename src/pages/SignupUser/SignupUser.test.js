// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import SignupUser from 'pages/SignupUser';

// test SignupUser component
describe('SignupUser Component', () => {
  it('matches the snapshot', () => {
    const component = shallow(<SignupUser />);
    expect(component).toMatchSnapshot();
  });
});
