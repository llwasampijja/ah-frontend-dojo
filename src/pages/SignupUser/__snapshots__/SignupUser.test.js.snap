// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupUser Component matches the snapshot 1`] = `
<div>
  <SignupForm
    confirmPassword=""
    email=""
    onChangeHandler={[Function]}
    onSubmitHandler={[Function]}
    password=""
    username=""
  />
</div>
`;
